paperqa/agents/env.py

```diff
@@ -162,19 +162,31 @@ async def step(
     ) -> tuple[list[Message], float, bool, bool]:
         self.state.session.add_tokens(action)  # Add usage for action if present
 
-        # If the action has empty tool_calls, the agent can later take that into account
-        msgs = cast(
+        valid_action, invalid_action = self.filter_invalid_tool_calls(action)
```

Comment:
Okay so, based on internal discussions today, I think we should check if `not action.tool_calls`, and return `done` if so

```python
if not action.tool_calls:
    return Message(content="Agent chose to finish."), 0, GenerateAnswer.FAILED_TO_ANSWER, False
```

Reply:
Let's save this for another PR happening after this one, I can take care of it

Reply:
Handled this separately in https://github.com/Future-House/paper-qa/pull/671